purchase_fiscal_position_update/models/purchase.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2011-2014 Julius Network Solutions SARL <contact@julius.fr>
+# @author Mathieu Vatel <mathieu _at_ julius.fr>
+# © 2014-2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.onchange('fiscal_position_id')
+    def fiscal_position_change(self):
```

Comment:
For me it's not, because any modification in the onchange code via inherited modules is not reflected here, so you will need glue modules. It's also not too heavy working on cached recordset. But it's needed a change in the approach (it seems awkward, but it's the way to go to keep the best compatibility, and it's fast):
```
for line in self.order_line:
    temp_line = line.new(line._convert_to_write(line._cache))
    temp_line.onchange_product_id()
    line.taxes_id = temp_line.taxes_id
```